File: Recursion/python/sum_n_natural_numbers.py
# TIME COMPLEXITY : O(n)
# SPACE COMPLEXITY: O(n)

def sum_n(n):
    if n == 0 :
        return 0
    else:
        return sum_n(n-1) + n


if __name__ == "__main__":
    print(sum_n(10))

File: Recursion/c/sum_n_natural_numbers.c
// TIME COMPLEXITY : O(n)
// SPACE COMPLEXITY: O(n)

#include <stdio.h>

int sum(int n)
{
    if (n==0)
        return 0;
    return sum(n-1) + n;
}

int main()
{
    int x;
    x = sum(10);
    printf("%d\n", x);
    return 0;
}